### PR TITLE
feat(core) : added support for send on cashaddr

### PR DESCRIPTION
**Pull request by mukeshsp**
Jira Ticket : https://bitgoinc.atlassian.net/browse/BG-35557

### CODE REVIEW

File: modules/core/src/v2/coins/abstractUtxoCoin.ts
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/abstractUtxoCoin.ts:447]: are the addresses for `actualOutputs` always supposed to be in canonical form?

if so, please call `findMissingOutputs()` with `actualOutputs` mapped to the canonical address format

if you do that, the code changes for this method are not required

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
please change the PR to use `feat(core)` and remove the use of the word "token" as it has a specific meaning that's unrelated to this change

File: modules/core/src/v2/coins/abstractUtxoCoin.ts
Comment (mukeshsp) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts:447]: it is not always be in cashaddr format, if user can pass base58 or cashaddr. So if user pass the base58 that will be picked up by the first if condition or for cashaddr we need to convert base58 to cashaddr and then check the condition.
Also we are updating expectedOutputs address with cashaddr so we can reuse for other function call.

Comment (OttoAllmendinger) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts:447]: It is much simpler if we convert the addresses on both sides of the equation to the canonical format first.

Comment (OttoAllmendinger) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts:447]: As far as I understand, `expectedOutputs` will always have `base58` addresses and `actualOutputs` will have either cashaddr or base58. If you pass `expectedOutputs` with addresses in canonical form this method is much simpler.

Comment (mukeshsp) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts]: got your point.  I updated the code as expectedOutputs can have cashaddr and base58 so we are converting into base58 for handling the flow.

File: modules/core/src/v2/coins/abstractUtxoCoin.ts
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/abstractUtxoCoin.ts:538]: 1. You can use `this.getFamily()` and `this.canonicalAddress()` instead
2. You should not mutate the object. Instead do this

```
const expectedOutputs = _.get(...).map((output) => {
  if (!isBch) {
    return output;
  }
  return { ...output, address: (this as Bch).canonicalAddress(output.address) };
});
```

Comment (OttoAllmendinger) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts:538]: we should actually add this method to AbstractBaseCoin
```
canonicalAddress(address: string): string {
  return address;
}
```

then you can skip the `as BCH` casts and the conditional

Comment (mukeshsp) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts]: yes, Thanks for suggestion. I updated the all the changes as suggested.

File: modules/core/src/v2/coins/abstractUtxoCoin.ts
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/abstractUtxoCoin.ts:534]: You can remove this. The `canonicalAddress` will work on any bch address.

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/core/src/v2/coins/abstractUtxoCoin.ts
Comment (mukeshsp) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts]: yes, updated this one too.

**Review by OttoAllmendinger — APPROVED** [review]
please wait for review by @tylerlevine

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**
